com.unity.render-pipelines.universal/Runtime/UniversalRenderPipeline.cs

```diff
@@ -22,14 +23,181 @@ public LightweightRenderPipeline(LightweightRenderPipelineAsset asset)
 
 namespace UnityEngine.Rendering.Universal
 {
+    // A collection of profiling samplers, cache, helpers and convenience functions for Universal profiling.
+    // Keeps most of the profiling related things organized in one place for editability.
+    // Static global state, mostly read/sometimes write, non-thread safe (like most things in Universal).
+    //
+    // Example:
+    // // Cached CPU
+    // using var profScope = UniversalProfilingCache.GetCPUScope("example name");
+    // // Static GPU
+    // using( UniversalProfilingCache.GetGPUScope( cmd, UniversalProfilingScope.Pipeline.ExampleStaticSampler ) ) {}
+    internal static class UniversalProfilingCache
+    {
+        private static Dictionary<string, ProfilingSampler> m_StringSamplerCache = new Dictionary<string, ProfilingSampler>();
+        private static Dictionary<int, ProfilingSampler>    m_IntSamplerCache    = new Dictionary<int, ProfilingSampler>();
+        private static Dictionary<int, ProfilingSampler>    m_CameraSamplerCache = new Dictionary<int, ProfilingSampler>();
+
+        // A static (fast) fallback profiling name.
+        public static readonly ProfilingSampler m_UnknownSampler = new ProfilingSampler("Unknown");
+
+        static UniversalProfilingCache()
+        {
+            // TODO: with Net 5.0 support
+            // m_SamplerCache.EnsureCapacity(64);
+        }
+
+        // Creates a ProfilingScope for immediate CPU profiling only.
+        // "Inl_" CPU time outside of any command buffers.
+        public static ProfilingScope GetCPUScope(string scopeName)
```

Comment:
Should have generic interface for Enums and an explicit api for perf test tracking enum.